Global key support (#170)
seek-oss/vocab: README.md

[tags]: https://support.phrase.com/hc/en-us/articles/5822598372252-Tags-Strings-
[configuration]: #Configuration

#### Global key

`vocab push` and `vocab pull` can support global keys mapping. When you want certain translations to use a specific/custom key in Phrase, add the `global-key` to the structure.

```jsonc
// translations.json
{
  "Hello": {
    "message": "Hello",
    "global-key": "hello"
  },
  "Goodbye": {
    "message": "Goodbye",
    "global-key": "app.goodbye.label"
  }
}
```

In the above example,

- `vocab push` will push the `hello` and `app.goodbye.label` keys to Phrase.
- `vocab pull` will pull translations from Phrase and map them to the `hello` and `app.goodbye.label` keys.

##### Error on no translation for global key

> We can throw error if Phrase does not have translation that matches a global key. This can be done by passing `--error-on-no-global-key-translation` flag to `vocab pull`. E.g.

Reviewer: ```suggestion
An error can be thrown if Phrase doesn't have translation that matches a global key. This can be done by passing `--error-on-no-global-key-translation` flag to `vocab pull`.
```

**suggestion:** might be worth refactoring the sentence to remove the ambiguous "we" (who's we? vocab? the vocab team?).

Author: @rstacruz That's a good call-out, appreciate it

> Vocab can be configured to throw an error if Phrase doesn't have translation that matches a global key. This can be done by passing `--error-on-no-global-key-translation` flag to `vocab pull`.

Reviewer 2: ```suggestion
By default, `vocab pull` will not error if a translation is missing in Phrase for a translation with a global key.
If you want to throw an error in this situation, pass the `--error-on-no-global-key-translation` flag:
```
